src/runtime/contrib/tensorrt/tensorrt_runtime.cc

```diff
@@ -178,7 +178,14 @@ class TensorRTRuntime : public JSONRuntimeBase {
    */
   void BuildEngine() {
     batch_size_ = data_entry_[input_var_eid_[0]]->shape[0];
-    if (trt_engine_cache_.count(std::make_pair(symbol_name_, batch_size_))) return;
+    if (trt_engine_cache_.count(std::make_pair(symbol_name_, batch_size_))) {
+      TensorRTEngineAndContext& engine_and_context =
+          trt_engine_cache_.at(std::make_pair(symbol_name_, batch_size_));
+      size_t binding_num = engine_and_context.engine->getNbBindings();
+      if (engine_and_context.device_buffers.size() == binding_num) {
```

Comment:
This could be `!engine_and_context.device_buffers.empty()` instead, it maybe communicates the purpose of this check better.

Reply:
Use empty() instead